### ISSUE TITLE
How do you decide the denominator 512.0 for lr scaler?

### ISSUE THREAD

**Issue by amirhfarzaneh** [facebookresearch/deit#65]
Hi there,

I wonder how do you decide on the denominator [here](https://github.com/facebookresearch/deit/blob/ab5715372db8c6cad5740714b2216d55aeae052e/main.py#L304)? You have currently set it as 512.0:

`linear_scaled_lr = args.lr * args.batch_size * utils.get_world_size() / 512.0`

**Comment by TouvronHugo** [comment]
Hi @amirhfarzaneh ,
It is an arbitrary choice, this scaling strategy is commonly used to train convents. 
Best,
Hugo

**Comment by amirhfarzaneh** [comment]
> Hi @amirhfarzaneh ,
> It is an arbitrary choice, this scaling strategy is commonly used to train convents.
> Best,
> Hugo

Great! thank you for the clarification.

**Comment by bartmch** [comment]
Hi, is the denominator really an arbitrary choice? Scaling learning rates are done to make sure the gradients are averaged correctly in DDP, right? So shouldn't '''args.batch_size * utils.get_world_size() > 1'''? In other words, the denominator cannot be arbitrary.
I'm not 100% sure hence the question marks.
